ansible_collections/arista/avd/roles/eos_designs_to_containerlab/templates/topology.yml.j2

```diff
@@ -0,0 +1,73 @@
+{# HEADER -- LOGIC implementation #}
+{% set node_types = dict() %}
+{% for node_type_key in node_type_keys | arista.avd.default(['spine', 'l3leaf', 'l2leaf']) | arista.avd.natural_sort %}
+{%     if hostvars[inventory_hostname][node_type_key] is arista.avd.defined %}
+{%         do node_types.update({node_type_key: hostvars[inventory_hostname][node_type_key]}) %}
+{%     endif %}
+{% endfor %}
+{# CONTAINERLABS configuration rendering #}
+---
+name: {{ fabric_name }}
+
+mgmt:
+  network: 'mgmt_{{fabric_name}}'
+  ipv4_subnet: {{ mgmt_network_v4 }}
+
+{# NODES definition #}
+topology:
+  nodes:
+{% for node_type in node_types | arista.avd.natural_sort %}
+{# --------------------- NODE GROUPS LIST --------------------- #}
```

Comment:
This approach will only work if the target for the play knows about all nodes. Since this role should run after 
`eos_designs`, I would prefer if we could read some of the generated data, instead of having to maintain the datamodel here as well.
I believe most of it should be in the structured_config including peer-switch information.

Reply:
Considering this role will be only for CI, local testing, it would be part of a specific playbook meaning the previous playbook will have to save debug files to make facts available in another play or playbook.

So user experience would be more complex than reading eos_designs inputs.

Reply:
I meant reading structured_configuration instead. So we can change eos_designs freely without affecting this one.

Reply:
Using `structured_config` files will create constraint when `eos_cli_config_gen` is similar to the current relationship with `eos_designs` and will also make rendering more complex:

- Run task for one host.
- Load `structured_config` for all hosts with a per-host approach to saving data.
- Read `structured_config` to generate file.

Where the current implementation just reads existing inventory variables.

> Note: this PR is only for the eos_designs approach